ui/pages/confirmations/components/confirm/footer/acknowledge/acknowledge.tsx

```diff
@@ -13,8 +16,11 @@ export function Acknowledge(props: AcknowledgeProps) {
   const t = useI18nContext();
   const isUpgradeTransaction = useIsUpgradeTransaction();
   const { isAcknowledged, onAcknowledgeToggle } = props;
+  const dismissSmartAccountSuggestionEnabled = useSelector(
+    getDismissSmartAccountSuggestionEnabled,
+  );
 
-  if (!isUpgradeTransaction) {
+  if (!isUpgradeTransaction || dismissSmartAccountSuggestionEnabled) {
```

Comment:
Unfortunately we can't just hide the acknowledge checkbox as the transaction would still perform the upgrade once approved.

If we instead check this here:

https://github.com/MetaMask/metamask-extension/blob/main/app/scripts/lib/transaction/eip5792.ts#L187

Then the middleware would reject the request outright since the user has said they never want to upgrade.

Plus technically we should also check if the user is already upgraded and skip this check, but that can be done in a separate ticket as I have pending changes in the transaction controller to get that information.

Reply:
oh my bad, I fixed this

Reply:
Let me create task for 
> Plus technically we should also check if the user is already upgraded and skip this check, but that can be done in a separate ticket as I have pending changes in the transaction controller to get that information.